install/update_0803_083.php

```diff
@@ -990,19 +1090,34 @@ function update0803to083() {
    $migration->displayMessage(sprintf(__('Data migration - %s'), 'various cleaning DB'));
 
    // Clean ticket satisfactions
-   $query = "DELETE
-             FROM `glpi_ticketsatisfactions`
-             WHERE `glpi_ticketsatisfactions`.`tickets_id` NOT IN (SELECT `glpi_tickets`.`id`
-                                                                   FROM `glpi_tickets`)";
-   $DB->queryOrDie($query, "0.83 clean glpi_ticketsatisfactions");
+   $DB->deleteOrDie("glpi_ticketsatisfactions", [
+         'NOT' => [
+            'glpi_ticketsatisfactions.tickets_id' => new \QuerySubQuery([
+               'SELECT' => "glpi_tickets.id",
+               'FROM'   => "glpi_tickets"
+            ])
+         ]
+      ],
+      "0.83 clean glpi_ticketsatisfactions"
+   );
 
    // Clean unused slalevels
-   $query = "DELETE
-             FROM `glpi_slalevels_tickets`
-             WHERE (`glpi_slalevels_tickets`.`tickets_id`, `glpi_slalevels_tickets`.`slalevels_id`)
-                  NOT IN (SELECT `glpi_tickets`.`id`, `glpi_tickets`.`slalevels_id`
-                          FROM `glpi_tickets`)";
-   $DB->queryOrDie($query, "0.83 clean glpi_slalevels_tickets");
+   $DB->deleteOrDie("glpi_slalevels_tickets", [
+         'NOT' => [
+            'glpi_slalevels_tickets.tickets_id' => new \QuerySubQuery([
+               'SELECT' => "glpi_tickets.id",
+               'FROM'   => "glpi_tickets"
+            ])
+         ],
+         'NOT' => [
```

Comment:
This won't work. Yoou have two keys with the same name, php will retrieve only one.

Reply:
I've replaced it by a QueyExpression to get the exact same behavior.